pkgs/development/python-modules/muscima/default.nix

```diff
@@ -33,14 +33,18 @@ buildPythonPackage {
     matplotlib
   ];
 
-  nativeCheckInputs = [
-    pytestCheckHook
-  ];
+  separateChecks = true;
 
-  disabledTestPaths = [
-    # They hard-code the path to the dataset and expect you to edit the test to update it to your value
-    "test/test_dataset.py"
-  ];
+  checkArgs = {
```

Comment:
Having to migrate all packages to this pattern sounds like major pain point. Can't we fish out the attrs we know are required for testing? Worst case we miss some and cause some cache misses which we can later improve?

Reply:
You don't actually need to pass `nativeCheckInputs`/`disabledTestPaths` via `checkArgs`. Those are fished out already.
We should indeed have a more exhaustive list of test related attrs.

https://github.com/NixOS/nixpkgs/pull/327264 is a good addition to this PR. It should probably go in before this.

Reply:
https://github.com/NixOS/nixpkgs/pull/272177#discussion_r1678731009